src/main/java/com/netgrif/application/engine/workflow/service/DataService.java

```diff
@@ -1164,4 +1166,12 @@ public void validateCaseRefValue(List<String> value, List<String> allowedNets) t
             }
         });
     }
+
+    public void validateTaskRefValue(List<String> value, String restrictedTaskId) throws IllegalArgumentException {
+        value.forEach(taskId -> {
+            if (taskId.equals(restrictedTaskId)) {
+                throw new IllegalArgumentException(String.format("Task with id '%s' cannot be added to task ref, since it is an task which displaying task ref", restrictedTaskId));
```

Comment:
possibly `"Task with id '%s' cannot be added to task ref, since it is a task which is displaying task ref"`.